feat: add radiantv2 protocol logics

LuPoYi opened the pull request:
# Pull Request Template

---

## Protocol Information

- **Name:** Bob Lu
- **Twitter:**
- **Website:**
- **Github:** https://github.com/LuPoYi
- **Developer Docs:**
- **Are you a member of the protocol?** No

## Checklist:

- [x] Added export `./{protocolId}` in `src/logics/index.ts`
- [x] Exported all Logics in `src/logics/{protocolId}/index.ts`
- [x] Ensured that all Logics have unit tests
- [x] Ensured that all Logics have integration tests

## Additional Notes

Temporarily skipping the Radiant v2 flashloan integration test due to an issue with the current RadiantV2FlashLoanCallback contract.

chouandy reviewed (approved):
LGTM

Jeff-CCH commented on test/logics/radiant-v2/borrow.test.ts:
```diff
@@ -0,0 +1,73 @@
+    chainId = await getChainId();
+    const [, user1, user2] = await hre.ethers.getSigners();
+    users = [user1, user2];
+    await claimToken(chainId, user1.address, radiantv2.arbitrumTokens.USDC, '20000');
```
Should we follow the same style to use the `arbitrumTokens` from test-helpers?

LuPoYi replied:
I think the `arbitrumTokens` from test-helpers are mainly for the most common tokens like `USDC`, `USDT`, and `WETH`. 

However, there might be cases where we need to include specific tokens (such as `aToken` or `rToken`) for testing that are not included in `arbitrumTokens` from test-helpers. In such cases, I believe it would be more appropriate to use `radiantv2.arbitrumTokens.USDC` when we have to get `token`.

Jeff-CCH commented on test/logics/radiant-v2/borrow.test.ts:
```diff
@@ -0,0 +1,73 @@
+      userIndex: 0,
+      deposit: new common.TokenAmount(radiantv2.arbitrumTokens.USDC, '5000'),
+      output: new common.TokenAmount(radiantv2.arbitrumTokens.WETH, '1'),
+      interestRateMode: radiantv2.InterestRateMode.variable,
```
Should we add tests for `InterestRateMode.stable`?

LuPoYi replied:
Due to Radiant v2 only support stable mode (on UI and their doc), the stable mode tests here may not be essential

Jeff-CCH commented on test/logics/radiant-v2/deposit.test.ts:
```diff
@@ -0,0 +1,89 @@
+      }
+
+      // 3. build router logics
+      const routerLogics: core.IParam.LogicStruct[] = [];
```
Suggest update `IParam` since the name has been changed on contract side

LuPoYi replied:
Could you give me more information about this? I currently not familiar with the code on contract side

LuPoYi replied:
updated.

Jeff-CCH commented on test/logics/radiant-v2/flash-loan.test.ts:
```diff
@@ -0,0 +1,82 @@
+import * as utility from 'src/logics/utility';
+import * as utils from 'test/utils';
+
+// TODO: skip this because current FlashLoanCallback got some issue
```
What kind of problem do we have now?

LuPoYi replied:
Because the current version of RadiantV2FlashLoanCallback is developed based on the Router version with `defaultReferral`, but the currently deployed Router version does not have `defaultReferral`, RadiantV2FlashLoanCallback will revert at the end.

But we deployed the new RadiantV2FlashLoanCallback contract, so I update the new contract address and unskip this test

Jeff-CCH commented on test/logics/radiant-v2/repay.test.ts:
```diff
@@ -0,0 +1,119 @@
+      userIndex: 0,
+      deposit: new common.TokenAmount(radiantv2.arbitrumTokens.USDC, '5000'),
+      borrow: new common.TokenAmount(radiantv2.arbitrumTokens.ETH, '1'),
+      interestRateMode: radiantv2.InterestRateMode.variable,
```
same as the add stable question

Jeff-CCH commented:
I found the arbitrum tests were not run in the e2e test. Could you help find out why?
Also, could you help to find out why the e2e test failed? tks.

Jeff-CCH commented:
Found another typo: `USD//C` in `src/logics/uniswap-v3/tokens/data/mainnet.json`. Could you help fix this too? tks.

LuPoYi commented:
Fixed all e2e test!

The reason arbitrum tests weren't run is because of the command `yarn run test:e2e:mainnet && yarn run test:e2e:arbitrum`. If e2e:mainnet encounters errors, it won't proceed with the subsequent command.

The failure in the mainnet compound-v2 claim test occurred because we were using legacy tokens, and only DAI currently receives COMP rewards. so, I have updated them to use DAI.

Jeff-CCH commented on package.json:
```diff
@@ -20,7 +20,7 @@
-    "test:e2e": "yarn run test:e2e:mainnet && yarn run test:e2e:polygon && yarn run test:e2e:arbitrum",
+    "test:e2e": "yarn run test:e2e:mainnet && yarn run test:e2e:arbitrum",
     "test:e2e:mainnet": "env-cmd -f .env.mainnet hardhat test --grep 'mainnet'",
     "test:e2e:polygon": "env-cmd -f .env.polygon hardhat test --grep 'polygon'",
```
If we don't have any polygon tests, is it a good idea to remove this line and `.env.polygon`?

LuPoYi replied:
Fixed it. My mistake, I removed it during testing and forgot to add it back.

Jeff-CCH commented on src/logics/compound-v2/scripts/get-tokens-data.ts:
```diff
@@ -12,8 +12,8 @@ const cTokenAddresses = [
-  '0x158079Ee67Fce2f58472A96584A73C7Ab9AC95c1', // cREP
-  '0xF5DCe57282A584D2746FaF1593d3121Fcac444dC', // cSAI
+  // '0x158079Ee67Fce2f58472A96584A73C7Ab9AC95c1', // cREP Legacy
+  // '0xF5DCe57282A584D2746FaF1593d3121Fcac444dC', // cSAI Legacy
```
Is it better to remove these lines rather than comment them out? We could add them back when we need them again.

LuPoYi replied:
Sure, and I don't think we need to add them back, so I've removed those lines.

Jeff-CCH commented on src/logics/compound-v2/tokens/data/cTokens.json:
```diff
@@ -55,13 +55,6 @@
     "symbol": "cMKR",
     "name": "Compound Maker"
   },
-  "cREP": {
```
@chouandy I found this file did not have `cWBTC2` info. Do we need to add it back? (I was checking the token info with `src/logics/compound-v2/scripts/get-tokens-data.ts`)

chouandy replied:
I think it's OK. But to address unrelated questions or topics, it's often more appropriate to create a new issue in the repository. This helps keep discussions organized and ensures that issues and PRs are dedicated to their respective purposes. cc @zodahu

Jeff-CCH replied:
Okay. Will create another ticket to track this issue.

Jeff-CCH reviewed (approved):
LGTM
